spynnaker/pyNN/models/recording_common.py

```diff
@@ -101,12 +111,59 @@ def _set_spikes_recording(self):
                 "This population does not support the recording of spikes!")
         self._population._vertex.set_recording_spikes()
 
-    def _get_recorded_variable(self, variable):
+    @property
+    def sampling_interval(self):
+        """ forced by the public nature of pynn variables
+
+        :return:
+        """
+
+        return self._sampling_interval
+
+    @sampling_interval.setter
+    def sampling_interval(self, new_value):
+        """ forced by the public nature of pynn variables
+
+        :param new_value: new value for the sampling_interval
+        :return: None
+        """
+        self._sampling_interval = new_value
+
+    @staticmethod
+    def pynn7_format(data, ids, sampling_interval, data2=None):
+        n_machine_time_steps = len(data)
+        n_neurons = len(ids)
+        column_length = n_machine_time_steps * n_neurons
+        times = [i * sampling_interval / 1000 for i in
+                 xrange(0, n_machine_time_steps)]
+        if data2 is None:
+            pynn7 = numpy.empty((column_length, 3))
+        else:
+            pynn7 = numpy.empty((column_length, 4))
+        pynn7[:, 0] = numpy.repeat(ids, n_machine_time_steps, 0).\
+            reshape(1, column_length)
+        pynn7[:, 1] = numpy.tile(times, n_neurons).\
+            reshape(1, column_length)
+        pynn7[:, 2] = numpy.transpose(data).reshape(1, column_length)
+        if data2 is not None:
+            pynn7[:, 3] = numpy.transpose(data2).reshape(1, column_length)
+        return pynn7
+
+    def _get_recorded_pynn7(self, variable):
+        if variable == "spikes":
+            data = self._get_spikes()
+
+        (data, ids, sampling_interval) = self._get_recorded_matrix(variable)
+        return self.pynn7_format(data, ids, sampling_interval)
+        (data, ids, sampling_interval) = self._get_recorded_matrix(variable)
+        return self.pynn7_format(data, ids, sampling_interval)
+
+    def _get_recorded_matrix(self, variable):
         """ method that contains all the safety checks and gets the recorded
-        data from the vertex
+        data from the vertex in P7NN0.7 format
```

Comment:
P7NN -> PyNN.  I thought this wasn't in PyNN 7 format though...

Reply:
Agreed